#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_use-sensible-browser.dpatch by Frank S. Thomas <frank@thomas-alfeld.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch is Debian specific. Don't use it for non Debian
## DP: packages. The program "sensible-browser", which is part of the
## DP: essential package debianutils, also uses the BROWSER variable, so
## DP: we don't loose functionality here.

@DPATCH@
diff -urNad boinc-5.4.9~/clientgui/hyperlink.cpp boinc-5.4.9/clientgui/hyperlink.cpp
--- boinc-5.4.9~/clientgui/hyperlink.cpp	2006-03-09 11:21:07.000000000 +0100
+++ boinc-5.4.9/clientgui/hyperlink.cpp	2006-05-06 16:10:25.000000000 +0200
@@ -166,77 +166,9 @@
 // private functions
 
 void wxHyperLink::ExecuteLink (const wxString &strLink) {
-    wxString cmd;
-    wxString strMimeType = wxEmptyString;
-    bool mime_type_found = false;
-
-    if      (strLink.StartsWith(wxT("http://")))
-        strMimeType = wxT("text/html");
-    else if (strLink.StartsWith(wxT("ftp://")))
-        strMimeType = wxT("text/html");
-    else if (strLink.StartsWith(wxT("mailto:")))
-        strMimeType = wxT("message/rfc822");
-    else
-        return;
-
-    wxFileType* ft = wxTheMimeTypesManager->GetFileTypeFromMimeType(strMimeType);
-    if (ft) {
-        if (ft->GetOpenCommand(&cmd, wxFileType::MessageParameters(strLink))) {
-#ifdef __WXMAC__
-            cmd.Replace(wxT("<"), wxEmptyString);
-            cmd.Prepend(wxT("open ")); 
-#else
-            cmd.Replace(wxT("file://"), wxEmptyString);
-#endif
-            mime_type_found = true;
-            ::wxExecute(cmd);
-        }
-        delete ft;
-    }
-
-#if defined(__WXGTK__) || defined(__WXMOTIF__)
-    if (!mime_type_found) {
-        cmd = ::wxGetenv(wxT("BROWSER"));
-        if(cmd.IsEmpty()) {
-            wxString strDialogTitle = wxEmptyString;
-            wxString strDialogMessage = wxEmptyString;
-
-            // %s is the application name
-            //    i.e. 'BOINC Manager', 'GridRepublic Manager'
-            strDialogTitle.Printf(
-                _("%s - Can't find web browser"),
-                wxGetApp().GetBrand()->GetApplicationName().c_str()
-            );
-
-            // 1st %s is the application name
-            //    i.e. 'BOINC Manager', 'GridRepublic Manager'
-            // 2nd %s is the URL that the browser is supposed to
-            //    open.
-            // 3rd %s is the application name
-            //    i.e. 'BOINC Manager', 'GridRepublic Manager'
-            strDialogMessage.Printf(
-                _("%s tried to display the web page\n"
-                "\t%s\n"
-                "but couldn't find a web browser.\n"
-                "To fix this, set the environment variable\n"
-                "BROWSER to the path of your web browser,\n"
-                "then restart the %s."),
-                wxGetApp().GetBrand()->GetApplicationName().c_str(),
-                strLink.c_str(),
-                wxGetApp().GetBrand()->GetApplicationName().c_str()
-            );
-
-            ::wxMessageBox(
-                strDialogMessage,
-                strDialogTitle,
-                wxOK | wxICON_INFORMATION
-            );
-        } else {
-            cmd += wxT(" ") + strLink;
-			::wxExecute(cmd,wxEXEC_ASYNC);
-        }
-    }
-#endif
+    wxString cmd = wxT("sensible-browser");
+    cmd += wxT(" ") + strLink;
+    ::wxExecute(cmd,wxEXEC_ASYNC);
 }
 
 const char *BOINC_RCSID_d587835b7e="$Id: hyperlink.cpp,v 1.11 2006/03/09 10:21:07 rwalton Exp $";
